packages/utilities/src/styled.tsx

```diff
@@ -97,21 +97,22 @@ export function styled<
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const cache = (styles.current && (styles.current as any).__cachedInputs__) || [];
-    if (!styles.current || customizedStyles !== cache[1] || props.styles !== cache[2]) {
+    const propStyles = props.styles;
```

Comment:
I would suggest adding a comment here explaining _why_ we are doing this, as closures are hard to reason about correctly and I could easily see this being accidentally reverted at some point in the future without some clear indicator that this is important for avoiding memory leaks.

Reply:
Good idea. Added a comment